privacyidea/lib/policymatch.py

```diff
@@ -0,0 +1,175 @@
+# -*- coding: utf-8 -*-
+#
+#  2019-07-19 Friedrich Weber <friedrich.weber@netknights.it>
+#             Add a high-level API for policy matching
+#
+#  License:  AGPLv3
+#  contact:  http://www.privacyidea.org
+#
+# This code is free software; you can redistribute it and/or
+# modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or any later version.
+#
+# This code is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+#
+# You should have received a copy of the GNU Affero General Public
+# License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+from privacyidea.lib.user import User
+from privacyidea.lib.error import ServerError
+from privacyidea.lib.policy import SCOPE
+
+
+class MatchingError(ServerError):
```

Comment:
Please add this to ``lib/policy.py``.

Reply:
We can do that! But as this PR is already pretty big, I'd like to move this to a later PR which would then also remove the remaining usages of ``match_policies``/``get_action_values``. What do you think?